contracts/parent-chain/espresso-migration/DeployAndInitEspressoSequencerInbox.s.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.9;
+
+import "forge-std/Script.sol";
+import "nitro-contracts/bridge/SequencerInbox.sol";
+import "nitro-contracts/bridge/ISequencerInbox.sol";
+
+
+contract DeployAndInitEspressoSequencerInbox is Script {
+    function run() external {
+        // Grab addresses from env
+        address bridgeAddr = vm.envAddress("BRIDGE_ADDRESS");
+        address reader4844Addr = vm.envAddress("READER_ADDRESS");
+        address espressoTeeVerifierAddr = vm.envAddress("ESPRESSO_TEE_VERIFIER_ADDRESS");
+     
+        // Grab the list of batch poster addresses valid for this rollup. This is a comma delimited list of addresses.
+        address[] memory batchPosters = vm.envAddress("BATCH_POSTER_ADDRS", ",");
+        address batchPosterManager = vm.envAddress("BATCH_POSTER_MANAGER");
+
+        // Grab any uints we need to initialize the contract from envAddress
+        uint256 deployerPrivateKey = vm.envUint("PRIVATE_KEY");
+        uint256 delayBlocks = vm.envUint("DELAY_BLOCKS");
+        uint256 futureBlocks = vm.envUint("FUTURE_BLOCKS"); 
+        uint256 delaySeconds = vm.envUint("DELAY_SECONDS");
+        uint256 futureSeconds = vm.envUint("FUTURE_SECONDS");
+        uint256 maxDataSize = vm.envUint("MAX_DATA_SIZE");
+        // Grab booleans we need from env
+        bool isUsingFeeToken = vm.envBool("IS_USING_FEE_TOKEN");
+
+        // initialize interfaces needed
+        IReader4844 reader = IReader4844(reader4844Addr);
+        IBridge bridge = IBridge(bridgeAddr);
+        // Start broadcast to deploy and initializer the sequencerInbox
+        vm.startBroadcast(deployerPrivateKey);
+        SequencerInbox sequencerInbox = new SequencerInbox(maxDataSize, reader, isUsingFeeToken);
```

Comment:
do we also need to change the chain config? https://github.com/EspressoSystems/nitro-contracts/blob/90037b996509312ef1addb3f9352457b8a99d6a6/src/precompiles/ArbOwner.sol#L94

Reply:
If we were to do that in one of the contracts it would be in the migration action contract. I was under the impression we wanted setting the chain config to be a discreet step in the process so that rollups could switch on posting to espresso after they have done all the migration steps. If we want the chain config switch to happen in the migration action, posting to espresso would begin as soon as the migration is done.

Reply:
Discussed, we will update the chainconfig in the migration action contract